packages/ag-charts-community/src/chart/axis/axisTickGenerator.ts

```diff
@@ -461,44 +562,131 @@ export class AxisTickGenerator<S extends Scale<D, number, TickInterval<S>>, D> {
                 rawTicks = this.filterTicks(previousTicks, tickCount);
                 break;
 
-            default:
-                rawTicks = scale.ticks(tickParams, niceDomain, visibleRange) ?? [];
+            case TickGenerationType.DROP_FIRST:
+                rawTicks = previousTicks.slice(1);
+                if (
+                    previousTicks.length > 2 &&
+                    previousPrimaryTicks != null &&
+                    previousTicks[0].valueOf() >= previousPrimaryTicks[0].valueOf() &&
+                    previousTicks[0].valueOf() < previousPrimaryTicks[1].valueOf()
+                ) {
+                    // Drop first primary tick if the raw tick was a primary tick
+                    primaryTicks = previousPrimaryTicks.slice(1);
+                } else {
+                    primaryTicks = previousPrimaryTicks;
+                }
+                interpolate = previousInterpolate;
+                timeInterval = previousTimeInterval;
+                break;
+
+            default: {
+                if (
+                    niceDomain.length > 0 &&
+                    (UnitTimeScale.is(scale) ||
+                        (primaryLabel != null && (TimeScale.is(scale) || OrdinalTimeScale.is(scale))))
+                ) {
+                    const dates = niceDomain as (Date | number)[];
+                    timeInterval = getTickTimeInterval(
+                        dates[0].valueOf(),
+                        dates[dates.length - 1].valueOf(),
+                        tickCount,
+                        minTickCount,
+                        maxTickCount
+                    );
+                }
+
+                if (primaryLabel != null) {
+                    const maxPrimaryTickRatio = 0.66;
+                    let firstLoop = true;
+                    while (
+                        timeInterval?.hierarchy &&
+                        (rawTicks == null || rawTicks.length > 5) &&
+                        (primaryTicks == null || primaryTicks.length > 5) &&
```

Comment:
Nit: Could we make some of these magic numbers more self-documenting (`0.66` and `5` here)?

Reply:
Updated

Reply:
Thanks!